app/database/user_item_store_compute_all_user_items.go

```diff
@@ -93,65 +93,79 @@ func (s *UserItemStore) ComputeAllUserItems() (err error) {
 
 			// For every object marked as 'processing', we compute all the characteristics based on the children:
 			//  - latest_activity_at as the max of children's
-			//  - tasks_with_help, tasks_tried, nbTaskSolved as the sum of children's field
-			//  - children_validated as the sum of children with validated == 1
+			//  - tasks_with_help, tasks_tried, nbTaskSolved as the sum of children's per-item maximums
+			//  - children_validated as the number of children items with validated == 1
 			//  - validated, depending on the items_items.category and items.validation_type
 			if updateStatement == nil {
 				const updateQuery = `
-					UPDATE users_items
-					LEFT JOIN (
+					UPDATE groups_attempts
+					LEFT JOIN LATERAL (
 						SELECT
-							MAX(children.latest_activity_at) AS latest_activity_at,
-							SUM(children.tasks_tried) AS tasks_tried,
-							SUM(children.tasks_with_help) AS tasks_with_help,
-							SUM(children.tasks_solved) AS tasks_solved,
-							SUM(validated) AS children_validated,
-							children.user_id AS user_id,
-							items_items.parent_item_id AS item_id
-						FROM users_items AS children 
-						JOIN items_items ON items_items.child_item_id = children.item_id
-						GROUP BY children.user_id, items_items.parent_item_id
-					) AS children_data
-						USING(user_id, item_id)
-					LEFT JOIN task_children_data_view AS task_children_data
-						ON task_children_data.user_item_id = users_items.id
+							parent_groups_attempts.id,
+							MAX(aggregated_children_attempts.latest_activity_at) AS latest_activity_at,
+							IFNULL(SUM(aggregated_children_attempts.tasks_tried), 0) AS tasks_tried,
+							IFNULL(SUM(aggregated_children_attempts.tasks_with_help), 0) AS tasks_with_help,
+							IFNULL(SUM(aggregated_children_attempts.tasks_solved), 0) AS tasks_solved,
+							IFNULL(SUM(aggregated_children_attempts.validated), 0) AS children_validated,
+							SUM(IFNULL(NOT aggregated_children_attempts.validated, 1)) AS children_non_validated,
+							SUM(items_items.category = 'Validation' AND IFNULL(NOT aggregated_children_attempts.validated, 1))
+								AS children_non_validated_categories,
+							MAX(aggregated_children_attempts.validated_at) AS max_validated_at,
+							MAX(IF(items_items.category = 'Validation', aggregated_children_attempts.validated_at, NULL)) AS max_validated_at_categories
+						FROM groups_attempts AS parent_groups_attempts
+						JOIN items_items ON(
+							parent_groups_attempts.item_id = items_items.parent_item_id
+						)
+						LEFT JOIN LATERAL (
```

Comment:
I started from this, but MySQL didn't use indexes in that case :(

Reply:
Worth a comment in the query to mention that.

Reply:
added the comment

Reply:
If we do `LEFT JOIN (SELECT ... GROUP BY ...) ON some_condition` (as suggested above), MySQL would group all the data (for all possible (group_id, item_id) pairs) for each of the outer rows. This means it would perform `(SELECT ... GROUP BY ...)` for each of groups_attempts rows.

Reply:
For the 'wrong results', I just mean that there is now way to get correct results effectively using 'LEFT JOIN' here.